test/conformance/conformance_parallel_reduce.cpp

```diff
@@ -174,3 +236,108 @@ TEST_CASE("parallel_[deterministic_]reduce and std::invoke") {
 }
 
 #endif
+
+template <typename Runner, typename... PartitionerContext>
+void test_vector_of_lists_rvalue_reduce_basic(const Runner& runner, PartitionerContext&&... args) {
+    constexpr std::size_t n_vectors = 10'000;
+
+    using inner_type = NeverCopyWrapper<int>;
+    using list_type = EmptyCopyList<inner_type>;
+    using vector_of_lists_type = std::vector<list_type>;
+
+    vector_of_lists_type vector_of_lists;
+
+    vector_of_lists.reserve(n_vectors);
+    for (std::size_t i = 0; i < n_vectors; ++i) {
+        list_type list;
+
+        list.insert(list.end(), inner_type{1});
+        list.insert(list.end(), inner_type{2});
+        list.insert(list.end(), inner_type{3});
+        list.insert(list.end(), inner_type{4});
+        list.insert(list.end(), inner_type{5});
+        vector_of_lists.emplace_back(std::move(list));
+    }
+
+    oneapi::tbb::blocked_range<std::size_t> range(0, n_vectors, n_vectors * 2);
+
+    auto reduce_body = [&](const decltype(range)& range_obj, list_type&& x) {
+        list_type new_list = std::move(x);
+
+        for (std::size_t index = range_obj.begin(); index != range_obj.end(); ++index) {
+            new_list.splice(new_list.end(), std::move(vector_of_lists[index]));
+        }
+        return new_list;
+    };
+
+    auto join_body = [&](list_type&& x, list_type&& y) {
+        list_type new_list = std::move(x);
+
+        new_list.splice(new_list.end(), std::move(y));
+        return new_list;
+    };
+
+    list_type result = runner(range, list_type{}, reduce_body, join_body, std::forward<PartitionerContext>(args)...);
+
+    list_type expected_result;
+
+    for (std::size_t i = 0; i < n_vectors; ++i) {
+        expected_result.insert(expected_result.end(), inner_type{1});
+        expected_result.insert(expected_result.end(), inner_type{2});
+        expected_result.insert(expected_result.end(), inner_type{3});
+        expected_result.insert(expected_result.end(), inner_type{4});
+        expected_result.insert(expected_result.end(), inner_type{5});
+    }
+
+    REQUIRE_MESSAGE(expected_result == result, "Incorrect reduce result");
+}
+
+struct ReduceRunner {
+    template <typename... Args>
+    auto operator()(Args&&... args) const -> decltype(oneapi::tbb::parallel_reduce(std::forward<Args>(args)...)) {
+        return oneapi::tbb::parallel_reduce(std::forward<Args>(args)...);
+    }
+};
+
+struct DeterministicReduceRunner {
+    template <typename... Args>
+    auto operator()(Args&&... args) const -> decltype(oneapi::tbb::parallel_deterministic_reduce(std::forward<Args>(args)...)) {
+        return oneapi::tbb::parallel_deterministic_reduce(std::forward<Args>(args)...);
+    }
```

Comment:
Is the trailing return type helpful versus just leave it to `auto` (or `decltype(auto)`)?

Reply:
We are forced to use trailing return type, because both `auto` with automatic deduction and `decltype(auto)` are C++14 features and we still need to support C++11.

Reply:
Gotcha. That's annoying.